api/database/mongooseIdPlugin.js

```diff
@@ -0,0 +1,7 @@
+export default function mongooseIdPlugin(schema, options) {
+  schema.options.toJSON.transform = function (doc, ret, options) {
+    delete ret._id; // Remove internal _id property before exposing it as JSON.
```

Comment:
Sounds good. If the JSON object can then be used seamlessly to update the document (with `id` instead of `_id`), then we should be fine.

But generally, I think we should better avoid having transformations in and out of the database, since it could create "mapping issues".